gcc build fails for CentOS 8 + RHEL 8 on ppc64le

Trying to build `gcc@7.3.0` on `CentOS 8` for `ppc64le` (`power8` host).

Using Spack @ `develop` (commit 17e4df1e4 - 16-March-2020) and the system-installed `gcc@8.3.1` as the base compiler

```
$> spack install gcc@7.3.0+strip target=ppc64le
...
==> Error: Failed to install gcc due to ChildError: ProcessError: Command exited with status 2:
    'make' '-j16'
1 warning found in build log:
     1419    checking for shared library run path origin... (cached) done
     1420    no
     1421    checking for VALGRIND_DISCARD in <memcheck.h>... checking for iconv... (cached) yes
     1422    checking for iconv declaration... (cached) install-shextern size_t iconv (iconv_t cd, char * *inbuf, size_t *inbytesleft, char * *outbuf, size_t
              *outbytesleft);
     1423    no
     1424    checking for multiarch configuration... auto
  >> 1425    configure: WARNING: fixed-point is not supported for this target, ignored
     1426    configure: creating ./config.status
     1427    checking whether make sets $(MAKE)... (cached) yes
     1428    checking for gawk... (cached) gawk
     1429    checking whether ln -s works... (cached) yes
     1430    checking whether ln works... (cached) yes
     1431    checking for powerpc64le-unknown-linux-gnu-ranlib... (cached) ranlib
```

Full build log: [gcc730-centos8-fail.txt](https://github.com/spack/spack/files/4340177/gcc730-centos8-fail.txt)

Likewise, trying to use Spack to build `gcc` on `RHEL8` for `ppc64le` (`power8` host) using the system installed `gcc@8.3.1` as the base compiler also fails. 

```
$> spack install gcc@9.3.0+strip target=ppc64le
...
==> Error: Failed to install gcc due to ChildError: ProcessError: Command exited with status 2:
    'make' '-j32'
5 errors found in build log:
     16864    checking for __atomic_fetch_add for size 2... mv -f .deps/snprintf-chk.Tpo .deps/snprintf-
              chk.Plo
     16865    mv -f .deps/vsprintf-chk.Tpo .deps/vsprintf-chk.Plo
     16866    config.status: creating Makefile
     16867    mv -f .deps/gets-chk.Tpo .deps/gets-chk.Plo
     16868    mv -f .deps/strncat-chk.Tpo .deps/strncat-chk.Plo
     16869    configure: loading cache ./config.cache
  >> 16870    configure: error: `CC' has changed since the previous run:
     16871    configure:   former value:  `/tmp/root/spack-stage/spack-stage-gcc-9.3.0-qglm6bea4vabbjfk4
              2g42ffxcyhe6xwk/spack-src/spack-build/./gcc/xgcc -B/tmp/root/spack-stage/spack-stage-gcc-9
              .3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/spack-src/spack-build/./gcc/ -B/opt/spack/opt/spack/
              linux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/powerpc64le-unkno
              wn-linux-gnu/bin/ -B/opt/spack/opt/spack/linux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4
              vabbjfk42g42ffxcyhe6xwk/powerpc64le-unknown-linux-gnu/lib/ -isystem /opt/spack/opt/spack/l
              inux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/powerpc64le-unknow
              n-linux-gnu/include -isystem /opt/spack/opt/spack/linux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-
              qglm6bea4vabbjfk42g42ffxcyhe6xwk/powerpc64le-unknown-linux-gnu/sys-include   -fchecking=1'
     16872    configure:   current value: `/tmp/root/spack-stage/spack-stage-gcc-9.3.0-qglm6bea4vabbjfk4
              2g42ffxcyhe6xwk/spack-src/spack-build/./gcc/xgcc -B/tmp/root/spack-stage/spack-stage-gcc-9
              .3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/spack-src/spack-build/./gcc/ -B/opt/spack/opt/spack/
              linux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/powerpc64le-unkno
              wn-linux-gnu/bin/ -B/opt/spack/opt/spack/linux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4
              vabbjfk42g42ffxcyhe6xwk/powerpc64le-unknown-linux-gnu/lib/ -isystem /opt/spack/opt/spack/l
              inux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/powerpc64le-unknow
              n-linux-gnu/include -isystem /opt/spack/opt/spack/linux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-
              qglm6bea4vabbjfk42g42ffxcyhe6xwk/powerpc64le-unknown-linux-gnu/sys-include   -fno-checking
              '
  >> 16873    configure: error: `CXX' has changed since the previous run:
     16874    configure:   former value:  ` /tmp/root/spack-stage/spack-stage-gcc-9.3.0-qglm6bea4vabbjfk
              42g42ffxcyhe6xwk/spack-src/spack-build/./gcc/xgcc -shared-libgcc -B/tmp/root/spack-stage/s
              pack-stage-gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/spack-src/spack-build/./gcc -nostdin
              c++ -L/tmp/root/spack-stage/spack-stage-gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/spack-s
              rc/spack-build/powerpc64le-unknown-linux-gnu/libstdc++-v3/src -L/tmp/root/spack-stage/spac
              k-stage-gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/spack-src/spack-build/powerpc64le-unkno
              wn-linux-gnu/libstdc++-v3/src/.libs -L/tmp/root/spack-stage/spack-stage-gcc-9.3.0-qglm6bea
              4vabbjfk42g42ffxcyhe6xwk/spack-src/spack-build/powerpc64le-unknown-linux-gnu/libstdc++-v3/
              libsupc++/.libs -B/opt/spack/opt/spack/linux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4va
              bbjfk42g42ffxcyhe6xwk/powerpc64le-unknown-linux-gnu/bin/ -B/opt/spack/opt/spack/linux-rhel
              8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/powerpc64le-unknown-linux-g
              nu/lib/ -isystem /opt/spack/opt/spack/linux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4vab
              bjfk42g42ffxcyhe6xwk/powerpc64le-unknown-linux-gnu/include -isystem /opt/spack/opt/spack/l
              inux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/powerpc64le-unknow
              n-linux-gnu/sys-include   -fchecking=1'
     16875    configure:   current value: ` /tmp/root/spack-stage/spack-stage-gcc-9.3.0-qglm6bea4vabbjfk
              42g42ffxcyhe6xwk/spack-src/spack-build/./gcc/xgcc -shared-libgcc -B/tmp/root/spack-stage/s
              pack-stage-gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/spack-src/spack-build/./gcc -nostdin
              c++ -L/tmp/root/spack-stage/spack-stage-gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/spack-s
              rc/spack-build/powerpc64le-unknown-linux-gnu/libstdc++-v3/src -L/tmp/root/spack-stage/spac
              k-stage-gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/spack-src/spack-build/powerpc64le-unkno
              wn-linux-gnu/libstdc++-v3/src/.libs -L/tmp/root/spack-stage/spack-stage-gcc-9.3.0-qglm6bea
              4vabbjfk42g42ffxcyhe6xwk/spack-src/spack-build/powerpc64le-unknown-linux-gnu/libstdc++-v3/
              libsupc++/.libs -B/opt/spack/opt/spack/linux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4va
              bbjfk42g42ffxcyhe6xwk/powerpc64le-unknown-linux-gnu/bin/ -B/opt/spack/opt/spack/linux-rhel
              8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/powerpc64le-unknown-linux-g
              nu/lib/ -isystem /opt/spack/opt/spack/linux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4vab
              bjfk42g42ffxcyhe6xwk/powerpc64le-unknown-linux-gnu/include -isystem /opt/spack/opt/spack/l
              inux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/powerpc64le-unknow
              n-linux-gnu/sys-include   -fno-checking'
  >> 16876    configure: error: in `/tmp/root/spack-stage/spack-stage-gcc-9.3.0-qglm6bea4vabbjfk42g42ffx
              cyhe6xwk/spack-src/spack-build/powerpc64le-unknown-linux-gnu/libstdc++-v3':
  >> 16877    configure: error: changes in the environment can compromise the build
  >> 16878    configure: error: run `make distclean' and/or `rm ./config.cache' and start over
     16879    config.status: creating backtrace-supported.h
     16880    mv -f .deps/ssp.Tpo .deps/ssp.Plo
     16881    /bin/sh ./libtool  --tag=CC   --mode=link /tmp/root/spack-stage/spack-stage-gcc-9.3.0-qglm
              6bea4vabbjfk42g42ffxcyhe6xwk/spack-src/spack-build/./gcc/xgcc -B/tmp/root/spack-stage/spac
              k-stage-gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/spack-src/spack-build/./gcc/ -B/opt/spa
              ck/opt/spack/linux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/powe
              rpc64le-unknown-linux-gnu/bin/ -B/opt/spack/opt/spack/linux-rhel8-ppc64le/gcc-8.3.1/gcc-9.
              3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/powerpc64le-unknown-linux-gnu/lib/ -isystem /opt/spac
              k/opt/spack/linux-rhel8-ppc64le/gcc-8.3.1/gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/power
              pc64le-unknown-linux-gnu/include -isystem /opt/spack/opt/spack/linux-rhel8-ppc64le/gcc-8.3
              .1/gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/powerpc64le-unknown-linux-gnu/sys-include
              -fno-checking -Wall  -g -O2 -version-info `grep -v '^#' /tmp/root/spack-stage/spack-stage-
              gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/spack-src/libssp/libtool-version` -Wl,--version
              -script=/tmp/root/spack-stage/spack-stage-gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/spack
              -src/libssp/ssp.map   -o libssp.la -rpath /opt/spack/opt/spack/linux-rhel8-ppc64le/gcc-8.3
              .1/gcc-9.3.0-qglm6bea4vabbjfk42g42ffxcyhe6xwk/lib/../lib64 ssp.lo gets-chk.lo memcpy-chk.l
              o memmove-chk.lo mempcpy-chk.lo memset-chk.lo snprintf-chk.lo sprintf-chk.lo stpcpy-chk.lo
               strcat-chk.lo strcpy-chk.lo strncat-chk.lo strncpy-chk.lo vsnprintf-chk.lo vsprintf-chk.l
              o
     16882    yes
     16883    libtool: link: ar rc .libs/libssp_nonshared.a  libssp_nonshared_la-ssp-local.o
     16884    libtool: link: ranlib .libs/libssp_nonshared.a
```

Full build log:  [gcc930-rhel8-fail.txt](https://github.com/spack/spack/files/4340379/gcc930-rhel8-fail.txt)

--- a/var/spack/repos/builtin/packages/gcc/package.py
+++ b/var/spack/repos/builtin/packages/gcc/package.py
@@ -119,6 +119,7 @@ class Gcc(AutotoolsPackage, GNUMirrorPackage):
     depends_on('isl@0.15:', when='@10:')
     depends_on('zlib', when='@6:')
     depends_on('zstd', when='@10:')
+    depends_on('diffutils', type='build')
     depends_on('iconv', when='platform=darwin')
     depends_on('gnat', when='languages=ada')
     depends_on('binutils~libiberty', when='+binutils', type=('build', 'link', 'run'))

